### ISSUE TITLE
callproc doesn't work with bytes input

### ISSUE THREAD

**Issue by Dobatymo** [pymssql/pymssql#598]
Python 3.6.6 (v3.6.6:4cf1f54eb7, Jun 27 2018, 03:37:03) [MSC v.1900 64 bit (AMD64)] on win32
tested pymssql version 2.1.3 (installed using the wheel from https://www.lfd.uci.edu/~gohlke/pythonlibs/#pymssql)
and pymssql version 2.1.4 (wheel from pip)

```
CREATE PROCEDURE [dbo].[usp_Broken]
  @data varbinary(max)
AS
RETURN 1
```

```
import pymssql

cursor = pymssql.connect(server, username, password, database).cursor()
print(cursor.callproc("usp_Broken", (b"asdf", )).returnvalue)
```
Results in:
```
Exception ignored in: 'pymssql.py2db_type'
KeyError: ('bytes',)
Traceback (most recent call last):
  File "C:\broken.py", line 12, in <module>
    print(cursor.callproc("usp_Broken", (b"asdf", )).returnvalue)
  File "pymssql.pyx", line 427, in pymssql.Cursor.callproc (pymssql.c:6629)
  File "_mssql.pyx", line 1458, in _mssql.MSSQLStoredProcedure.bind (_mssql.c:15508)
  File "_mssql.pyx", line 978, in _mssql.MSSQLConnection.convert_python_value (_mssql.c:11067)
_mssql.MSSQLDriverException: Unable to convert value
```

**Comment by AlexHagerman** [comment]
@Dobatymo this is strange. We probably need to review some of the proc code as I've seen other issues opened relating to the stored proc call. One thing that I find odd also is the `KeyError`. Can you set a breakpoint or profile sql server to see what value is being sent that can't convert?

**Comment by Dobatymo** [comment]
As far as I understand the code, this exception is thrown even before the call makes it to the database.

**Comment by Dobatymo** [comment]
I think I found the problem https://github.com/pymssql/pymssql/blob/891b20e29e4e247c17b202e8e34e5c739b6090ef/src/pymssql.pyx#L100-L110

`str` and `unicode` are the same on Python 3 (https://cython.readthedocs.io/en/latest/src/tutorial/strings.html). So the correct types to map would be `bytes` to `VARBINARY` and `unicode` to `VARCHAR`. That would work on Python 2 and 3 I think.

I cannot test it, because I cannot setup my environment for building it right now.

**Comment by termim** [comment]
Fixed in 50c632ef2bce448effefbebc51a66726b4a9c163
